File: Legacy-Folder/Backend/HRMSWebApi/HRMS.Domain/Entities/ResignationHistory.cs
using HRMS.Domain.Enums;

namespace HRMS.Domain.Entities
{
    public class ResignationHistory
    {
        public long Id { get; set; }
        public int ResignationId { get; set; }
        public ResignationStatus ResignationStatus { get; set; }
        public DateTime CreatedOn { get; set; }
        public string CreatedBy { get; set; }
        public EarlyReleaseStatus EarlyReleaseStatus { get; set; }
    }
}

File: hrms-backend/app/Models/ResignationHistory.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class ResignationHistory extends Model
{
    protected $table = 'resignation_history';
    protected $primaryKey = 'Id';
    public $timestamps = false;

    protected $fillable = [
        'ResignationId', 'ResignationStatus', 'EarlyReleaseStatus', 'CreatedOn', 'CreatedBy'
    ];

    protected $casts = [
        'CreatedOn' => 'datetime',
    ];

    // Relationship: History belongs to Resignation
    public function resignation()
    {
        return $this->belongsTo(Resignation::class, 'ResignationId', 'Id');
    }
}
